src/signature.c

```diff
@@ -121,6 +122,44 @@ int git_signature_dup(git_signature **dest, const git_signature *source)
 	return 0;
 }
 
+int git_signature_with_mailmap(
+	git_signature **dest,
+	const git_signature *source,
+	const git_mailmap *mailmap)
+{
+	git_signature *signature = NULL;
+	const char *name = NULL;
+	const char *email = NULL;
+
+	if (source == NULL)
+		goto on_error;
+
+	git_mailmap_resolve(&name, &email, mailmap, source->name, source->email);
+
+	signature = git__calloc(1, sizeof(git_signature));
+	if (!signature)
+		goto on_error;
```

Comment:
For these three cases, simply use `GITERR_CHECK_ALLOC`. Our current policy is to ignore memory leaks when we experience OOM-conditions

Reply:
Good to know! I was confused by existing consumers of GITERR_CHECK_ALLOC because they often seemed to leak memory, but that makes a lot of sense.